Add some Vale styles

stichbury opened the pull request:
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description
This PR adds styles to the Vale linter



## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

ankatiyar commented on .github/styles/Kedro/ukspelling.yml:
```diff
@@ -0,0 +1,25 @@
+    - seize
+    - size
+    - tailor
+    
```
Also add "DataCatalog" to the list of exceptions?

deepyaman commented on .github/styles/Kedro/words.yml:
```diff
@@ -0,0 +1,52 @@
+  'and/or': 'and|or|either or'
+  'back end': 'backend'
+  'bear in mind': 'keep in mind'
+  'culprit': 'cause'
```
```suggestion
  'culprit': 'cause'
  'data set': 'dataset'
```
:)

(Also, would apply to the "web site" rule, too--do you need to create a separate rule for the plural?)

stichbury commented:
> LGTM! My only concern would be if get a lot of warnings (not errors), to the point people end up ignoring them, because there are a lot of rules. I like it, but I'd be curious to see how many issues this come out with. :)

This is my worry too, which is why I want to try it out over the next sprint or two. I've already dropped the level from error->warning for most things and removed some of the less "wordy" words from `toowordy.yml`. If it's getting too annoying I'll revise it.

astrojuanlu reviewed (approved):
The rules look great! Can't wait to see the automatic comments on PRs 😄

ankatiyar reviewed (approved):
LGTM 🥇

DimedS reviewed (commented):
Great styles! With that linter, we'll all become professional English writers! 👍
